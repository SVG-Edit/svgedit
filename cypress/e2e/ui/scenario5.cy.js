import {
  visitAndApproveStorage
} from '../../support/ui-test-helper.js'

describe('use line tools of svg-edit', function () {
  before(() => {
    visitAndApproveStorage()
  })

  it('check tool_source_set', function () {
    cy.get('#tool_source').click({ force: true })
    cy.get('#svg_source_textarea')
      .type('{selectall}', { force: true })
      .type(`<svg width="640" height="480" xmlns="http://www.w3.org/2000/svg" xmlns:svg="http://www.w3.org/2000/svg">
      <g class="layer">
       <title>Layer 1</title>
       </g>
     </svg>`, { force: true, parseSpecialCharSequences: false })
    cy.get('#tool_source_save').click({ force: true })
    cy.svgSnapshot()
  })
  it('check tool_line', function () {
    cy.get('#tool_line')
      .click({ force: true })
    cy.get('#svgcontent')
      .trigger('mousemove', 200, 200, { force: true })
      .trigger('mousedown', 200, 200, { force: true })
      .trigger('mousemove', 250, 250, { force: true })
      .trigger('mouseup', { force: true })
    cy.svgSnapshot()
  })
  it('check tool_line_change_class', function () {
    cy.get('#svg_1').click({ force: true })
    cy.get('#elem_class').shadow().find('elix-input').eq(0).shadow().find('#inner').eq(0)
      .type('svg_1_class{enter}', { force: true })
    cy.get('#svg_1')
      .should('satisfy', ($el) => {
        const classList = Array.from($el[0].classList)
        return classList.includes('svg_1_class')
      })
  })
  it('check tool_line_change_id', function () {
    cy.get('#svg_1').click({ force: true }).click({ force: true })
    cy.get('#elem_id').shadow().find('elix-input').eq(0).shadow().find('#inner').eq(0)
      .type('_id{enter}', { force: true })
    cy.get('#svg_1_id')
      .should('satisfy', ($el) => {
        const classList = Array.from($el[0].classList)
        return classList.includes('svg_1_class')
      })
  })
  it('check tool_line_change_rotation', function () {
    cy.get('#svg_1_id').click({ force: true })
    for (let n = 0; n < 5; n++) {
      cy.get('#angle').shadow().find('elix-number-spin-box').eq(0).shadow().find('#upButton').eq(0)
        .click({ force: true })
    }
    cy.svgSnapshot()
  })
  it('check tool_line_change_blur', function () {
    cy.get('#svg_1_id').click({ force: true })
    for (let n = 0; n < 10; n++) {
      cy.get('#blur').shadow().find('elix-number-spin-box').eq(0).shadow().find('#upButton').eq(0)
        .click({ force: true })
    }
    cy.svgSnapshot()
  })
  it('check tool_line_change_opacity', function () {
    cy.get('#svg_1_id').click({ force: true })
    for (let n = 0; n < 10; n++) {
      cy.get('#opacity').shadow().find('elix-number-spin-box').eq(0).shadow().find('#downButton').eq(0)
        .click({ force: true })
    }
    cy.svgSnapshot()
  })
  it('check tool_line_delete', function () {
    cy.get('#svg_1_id').click({ force: true })
    cy.get('#tool_delete').click({ force: true })
    cy.svgSnapshot()
  })
  it('check tool_line_clone', function () {
    cy.get('#tool_line')
      .click({ force: true })
    cy.get('#svgcontent')
      .trigger('mousemove', 200, 200, { force: true })
      .trigger('mousedown', 200, 200, { force: true })
      .trigger('mousemove', 250, 250, { force: true })
      .trigger('mouseup', { force: true })
    cy.get('#svg_2').click({ force: true })
    cy.get('#tool_clone').click({ force: true })
    cy.svgSnapshot()
  })
  it('check tool_line_bring_to_back', function () {
    cy.get('#svg_2').click({ force: true })
    cy.get('#tool_move_bottom').click({ force: true })
    cy.svgSnapshot()
  })
  it('check tool_line_bring_to_front', function () {
    cy.get('#svg_2').click({ force: true })
    cy.get('#tool_move_top').click({ force: true })
    cy.svgSnapshot()
  })
  it('check tool_line_change_x_y_coordinate', function () {
    cy.get('#svg_2').click({ force: true })
    for (let n = 0; n < 25; n++) {
      cy.get('#line_x1').shadow().find('elix-number-spin-box').eq(0).shadow().find('#upButton').eq(0)
        .click({ force: true })
    }
    for (let n = 0; n < 25; n++) {
      cy.get('#line_y1').shadow().find('elix-number-spin-box').eq(0).shadow().find('#downButton').eq(0)
        .click({ force: true })
    }
    for (let n = 0; n < 25; n++) {
      cy.get('#line_x2').shadow().find('elix-number-spin-box').eq(0).shadow().find('#upButton').eq(0)
        .click({ force: true })
    }
    for (let n = 0; n < 25; n++) {
      cy.get('#line_y2').shadow().find('elix-number-spin-box').eq(0).shadow().find('#downButton').eq(0)
        .click({ force: true })
    }
    cy.svgSnapshot()
  })
  it('check tool_line_change_stroke_width', function () {
    cy.get('#svg_2').click({ force: true })
    for (let n = 0; n < 10; n++) {
      cy.get('#stroke_width').shadow().find('elix-number-spin-box').eq(0).shadow().find('#upButton').eq(0)
        .click({ force: true })
    }
    cy.svgSnapshot()
  })
  it('check tool_line_change_stoke_color', function () {
    cy.get('#svg_3').click({ force: true })
    cy.get('#stroke_color').shadow().find('#picker').eq(0).click({ force: true })
    cy.get('#stroke_color').shadow().find('#color_picker').eq(0)
      .find('#jGraduate_colPick').eq(0).find('#jPicker-table').eq(0)
      .find('.QuickColor').eq(9).click({ force: true })
    cy.get('#stroke_color').shadow().find('#color_picker').eq(0)
      .find('#jGraduate_colPick').eq(0).find('#jPicker-table').eq(0)
      .find('#Ok').eq(0).click({ force: true })
    cy.svgSnapshot()
  })
  it('check tool_line_align_to_page', function () {
    cy.get('#svg_3').click({ force: true })
    cy.get('#tool_position').shadow().find('#select-container').eq(0).click({ force: true })
    cy.get('#tool_position').find('se-list-item').eq(2).shadow().find('[aria-label="option"]').eq(0)
      .click({ force: true })
    cy.svgSnapshot()
  })
})
