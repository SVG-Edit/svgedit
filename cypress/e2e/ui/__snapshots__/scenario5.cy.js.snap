exports[`use all parts of svg-edit > check tool_source_set #0`] = `
<body>
  <svg
    width="640"
    height="480"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:svg="http://www.w3.org/2000/svg"
  >
    <g class="layer">
      <title>Layer 1</title>
    </g>
  </svg>
</body>
`;

exports[`use all parts of svg-edit > check tool_line #0`] = `
<body>
  <svg
    width="640"
    height="480"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:svg="http://www.w3.org/2000/svg"
  >
    <g class="layer">
      <title>Layer 1</title>
      <line
        fill="none"
        id="svg_1"
        opacity="0.5"
        stroke="#000000"
        stroke-width="5"
        x1="200"
        x2="450"
        y1="200"
        y2="450"
      >
        <animate
          attributeName="opacity"
          begin="indefinite"
          dur="0.2"
          fill="freeze"
          to="1"
        ></animate>
      </line>
    </g>
  </svg>
</body>
`;

exports[`use all parts of svg-edit > check tool_line_change_rotation #0`] = `
<body>
  <svg
    width="640"
    height="480"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:svg="http://www.w3.org/2000/svg"
  >
    <g class="layer">
      <title>Layer 1</title>
      <line
        class="svg_1_class"
        fill="none"
        id="svg_1_id"
        stroke="#000000"
        stroke-width="5"
        transform="rotate(25 325 325)"
        x1="200"
        x2="450"
        y1="200"
        y2="450"
      ></line>
    </g>
  </svg>
</body>
`;

exports[`use all parts of svg-edit > check tool_line_change_blur #0`] = `
<body>
  <svg
    width="640"
    height="480"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:svg="http://www.w3.org/2000/svg"
  >
    <g class="layer">
      <title>Layer 1</title>
      <line
        class="svg_1_class"
        fill="none"
        filter="url(#svg_1_id_blur)"
        id="svg_1_id"
        stroke="#000000"
        stroke-width="5"
        transform="rotate(25 325 325)"
        x1="200"
        x2="450"
        y1="200"
        y2="450"
      ></line>
    </g>
    <defs>
      <filter height="200%" id="svg_1_id_blur" width="200%" x="-50%" y="-50%">
        <feGaussianBlur in="SourceGraphic" stdDeviation="5"></feGaussianBlur>
      </filter>
    </defs>
  </svg>
</body>
`;

exports[`use all parts of svg-edit > check tool_line_change_opacity #0`] = `
<body>
  <svg
    width="640"
    height="480"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:svg="http://www.w3.org/2000/svg"
  >
    <g class="layer">
      <title>Layer 1</title>
      <line
        class="svg_1_class"
        fill="none"
        filter="url(#svg_1_id_blur)"
        id="svg_1_id"
        opacity="0.5"
        stroke="#000000"
        stroke-width="5"
        transform="rotate(25 325 325)"
        x1="200"
        x2="450"
        y1="200"
        y2="450"
      ></line>
    </g>
    <defs>
      <filter height="200%" id="svg_1_id_blur" width="200%" x="-50%" y="-50%">
        <feGaussianBlur in="SourceGraphic" stdDeviation="5"></feGaussianBlur>
      </filter>
    </defs>
  </svg>
</body>
`;

exports[`use all parts of svg-edit > check tool_line_delete #0`] = `
<body>
  <svg
    width="640"
    height="480"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:svg="http://www.w3.org/2000/svg"
  >
    <g class="layer">
      <title>Layer 1</title>
    </g>
  </svg>
</body>
`;

exports[`use all parts of svg-edit > check tool_line_clone #0`] = `
<body>
  <svg
    width="640"
    height="480"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:svg="http://www.w3.org/2000/svg"
  >
    <g class="layer">
      <title>Layer 1</title>
      <line
        fill="none"
        id="svg_2"
        opacity="0.25"
        stroke="#000000"
        stroke-width="5"
        x1="200"
        x2="450"
        y1="200"
        y2="450"
      >
        <animate
          attributeName="opacity"
          begin="indefinite"
          dur="0.2"
          fill="freeze"
          to="0.5"
        ></animate>
      </line>
      <line
        fill="none"
        id="svg_3"
        opacity="0.25"
        stroke="#000000"
        stroke-width="5"
        x1="220"
        x2="470"
        y1="220"
        y2="470"
      >
        <animate
          attributeName="opacity"
          begin="indefinite"
          dur="0.2"
          fill="freeze"
          id="svg_4"
          to="0.5"
        ></animate>
      </line>
    </g>
  </svg>
</body>
`;

exports[`use all parts of svg-edit > check tool_line_bring_to_back #0`] = `
<body>
  <svg
    width="640"
    height="480"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:svg="http://www.w3.org/2000/svg"
  >
    <g class="layer">
      <title>Layer 1</title>
      <line
        fill="none"
        id="svg_2"
        opacity="0.5"
        stroke="#000000"
        stroke-width="5"
        x1="200"
        x2="450"
        y1="200"
        y2="450"
      ></line>
      <line
        fill="none"
        id="svg_3"
        opacity="0.25"
        stroke="#000000"
        stroke-width="5"
        x1="220"
        x2="470"
        y1="220"
        y2="470"
      >
        <animate
          attributeName="opacity"
          begin="indefinite"
          dur="0.2"
          fill="freeze"
          id="svg_4"
          to="0.5"
        ></animate>
      </line>
    </g>
  </svg>
</body>
`;

exports[`use all parts of svg-edit > check tool_line_bring_to_front #0`] = `
<body>
  <svg
    width="640"
    height="480"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:svg="http://www.w3.org/2000/svg"
  >
    <g class="layer">
      <title>Layer 1</title>
      <line
        fill="none"
        id="svg_3"
        opacity="0.25"
        stroke="#000000"
        stroke-width="5"
        x1="220"
        x2="470"
        y1="220"
        y2="470"
      >
        <animate
          attributeName="opacity"
          begin="indefinite"
          dur="0.2"
          fill="freeze"
          id="svg_4"
          to="0.5"
        ></animate>
      </line>
      <line
        fill="none"
        id="svg_2"
        opacity="0.5"
        stroke="#000000"
        stroke-width="5"
        x1="200"
        x2="450"
        y1="200"
        y2="450"
      ></line>
    </g>
  </svg>
</body>
`;

exports[`use all parts of svg-edit > check tool_line_change_x_y_coordinate #0`] = `
<body>
  <svg
    width="640"
    height="480"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:svg="http://www.w3.org/2000/svg"
  >
    <g class="layer">
      <title>Layer 1</title>
      <line
        fill="none"
        id="svg_3"
        opacity="0.25"
        stroke="#000000"
        stroke-width="5"
        x1="220"
        x2="470"
        y1="220"
        y2="470"
      >
        <animate
          attributeName="opacity"
          begin="indefinite"
          dur="0.2"
          fill="freeze"
          id="svg_4"
          to="0.5"
        ></animate>
      </line>
      <line
        fill="none"
        id="svg_2"
        opacity="0.5"
        stroke="#000000"
        stroke-width="5"
        x1="225"
        x2="475"
        y1="175"
        y2="425"
      ></line>
    </g>
  </svg>
</body>
`;

exports[`use all parts of svg-edit > check tool_line_change_stroke_width #0`] = `
<body>
  <svg
    width="640"
    height="480"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:svg="http://www.w3.org/2000/svg"
  >
    <g class="layer">
      <title>Layer 1</title>
      <line
        fill="none"
        id="svg_3"
        opacity="0.25"
        stroke="#000000"
        stroke-width="5"
        x1="220"
        x2="470"
        y1="220"
        y2="470"
      >
        <animate
          attributeName="opacity"
          begin="indefinite"
          dur="0.2"
          fill="freeze"
          id="svg_4"
          to="0.5"
        ></animate>
      </line>
      <line
        fill="none"
        id="svg_2"
        opacity="0.5"
        stroke="#000000"
        stroke-width="15"
        x1="225"
        x2="475"
        y1="175"
        y2="425"
      ></line>
    </g>
  </svg>
</body>
`;

exports[`use all parts of svg-edit > check tool_line_change_stoke_color #0`] = `
<body>
  <svg
    width="640"
    height="480"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:svg="http://www.w3.org/2000/svg"
  >
    <g class="layer">
      <title>Layer 1</title>
      <line
        fill="none"
        id="svg_3"
        opacity="0.25"
        stroke="#bf5f00"
        stroke-width="5"
        x1="220"
        x2="470"
        y1="220"
        y2="470"
      >
        <animate
          attributeName="opacity"
          begin="indefinite"
          dur="0.2"
          fill="freeze"
          id="svg_4"
          to="0.5"
        ></animate>
      </line>
      <line
        fill="none"
        id="svg_2"
        opacity="0.5"
        stroke="#000000"
        stroke-width="15"
        transform="matrix(1 0 0 1 0 0)"
        x1="225"
        x2="475"
        y1="175"
        y2="425"
      ></line>
    </g>
  </svg>
</body>
`;

exports[`use all parts of svg-edit > check tool_line_align_to_page #0`] = `
<body>
  <svg
    width="640"
    height="480"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:svg="http://www.w3.org/2000/svg"
  >
    <g class="layer">
      <title>Layer 1</title>
      <line
        fill="none"
        id="svg_3"
        opacity="0.25"
        stroke="#bf5f00"
        stroke-width="5"
        x1="220"
        x2="470"
        y1="220"
        y2="470"
      >
        <animate
          attributeName="opacity"
          begin="indefinite"
          dur="0.2"
          fill="freeze"
          id="svg_4"
          to="0.5"
        ></animate>
      </line>
      <line
        fill="none"
        id="svg_2"
        opacity="0.5"
        stroke="#000000"
        stroke-width="15"
        transform="matrix(1 0 0 1 0 0)"
        x1="225"
        x2="475"
        y1="175"
        y2="425"
      ></line>
    </g>
  </svg>
</body>
`;
